src/contrib/cluster/Akka.Cluster.Tools/Singleton/OldestChangedBuffer.cs

```diff
@@ -135,65 +126,64 @@ private void SetupCoordinatedShutdown()
                 }
             });
         }
-
-        private readonly string _role;
-        private ImmutableSortedSet<Member> _membersByAge;
+        
         private ImmutableQueue<object> _changes = ImmutableQueue<object>.Empty;
 
         private readonly Cluster _cluster = Cluster.Get(Context.System);
-
-        private void TrackChanges(Action block)
-        {
-            var before = _membersByAge.FirstOrDefault();
-            block();
-            var after = _membersByAge.FirstOrDefault();
-
-            // todo: fix neq comparison
-            if (!Equals(before, after))
-                _changes = _changes.Enqueue(new OldestChanged(after?.UniqueAddress));
-        }
-
-        private bool MatchingRole(Member member)
-        {
-            return string.IsNullOrEmpty(_role) || member.HasRole(_role);
-        }
-
-        private void HandleInitial(ClusterEvent.CurrentClusterState state)
+        
+        private void HandleInitial(ClusterEvent.CurrentClusterState clusterState)
         {
             // all members except Joining and WeaklyUp
-            _membersByAge = state.Members
-                .Where(m => m.UpNumber != int.MaxValue && MatchingRole(m))
+            var newMembersByAge = clusterState.Members
+                .Where(m => m.UpNumber != int.MaxValue && State.MatchingRole(m))
                 .ToImmutableSortedSet(_memberAgeComparer);
+            
+           State = State with { MembersByAge = newMembersByAge };
+           
+           // compute the initial oldest - doesn't matter if it's not "safe" or not. That's our parent's problem.
```

Comment:
See comment for details - behavior is the same as before, just computed by the `State`